### PR TITLE
Add mahadzaryab1 to maintainers

### LINKED ISSUE #6264: Promote mahadzaryab1 to Maintainer
Mahad Zaryab (@mahadzaryab1) has continued his unstoppable streak of amazing pull requests (39!) and is now formally meets all the [requirements for becoming a maintainer](https://github.com/jaegertracing/jaeger/blob/main/GOVERNANCE.md#becoming-a-maintainer).

Examples of non-trivial PRs:
  * #5878
  * Normalizing/standardizing v2 configurations
    * #5937 
    * #5998
    * #5949 
    * #6042
    * #6060
    * #6080
  * Migrating to OTEL helpers for servers & tls
    * #6079
    * #6055
    * #6121 
  * Ongoing work on Storage V2 API
    * #6216 
    * #6221

I propose to promote him to full Jaeger Maintainer role.

### PR DESCRIPTION
**Pull request by yurishkuro**
## Which problem is this PR solving?
- Resolves #6264

## Description of the changes
- Add new maintainer
- Clean-up README

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -5,9 +5,20 @@ The current Maintainers Group for the Jaeger Project consists of:
 | [@albertteoh](https://github.com/albertteoh) | PackSmith | ALL | 
 | [@jkowall](https://github.com/jkowall) | Aiven | ALL |
 | [@joe-elliott](https://github.com/joe-elliott) | Grafana Labs | ALL |
+| [@mahadzaryab1](https://github.com/mahadzaryab1) | ? | ALL |
```
Comment (yurishkuro) [inline MAINTAINERS.md:8]: @mahadzaryab1 what should we put for your Employer?

File: RELEASE.md [hunk 2]
```diff
@@ -78,8 +78,9 @@ Here are the release managers for future versions with the tentative release dat
 
 | Version | Release Manager | Tentative release date |
 |---------|-----------------|------------------------|
-| 1.65.0  | @jkowall        | 8 January 2025         |
-| 1.66.0  | @yurishkuro     | 3 February 2025        |
-| 1.67.0  | @albertteoh     | 5 March 2025           |
-| 1.68.0  | @pavolloffay    | 5 April 2025           |
-| 1.69.0  | @joe-elliott    | 7 May   2025           |
\ No newline at end of file
+| 1.65.0  | @mahadzaryab1   | 8 January 2025         |
```
Comment (yurishkuro) [inline RELEASE.md:81]: @mahadzaryab1 I added you to the next release to try your hands on the release process, ok?

File: MAINTAINERS.md (hunk 1 above)
Comment (mahadzaryab1) [reply MAINTAINERS.md:8]: Bloomberg

File: RELEASE.md (hunk 2 above)
Comment (mahadzaryab1) [reply RELEASE.md:81]: yeah - sounds great! thank you

**Review by mahadzaryab1 — APPROVED** [review]
thank you so much!

**Pull request merged**